src/log/log.c

```diff
@@ -2272,7 +2277,7 @@ __wt_log_scan(WT_SESSION_IMPL *session, WT_LSN *lsnp, uint32_t flags,
 				    PRIu32 "/%" PRIu32,
 				    lsnp->l.file, lsnp->l.offset);
 			else
-				return (WT_NOTFOUND);
```

Comment:
The comment makes these returns sound deliberate. @sueloverso or @luke-pearson could you chime in as to whether this change is correct?

Reply:
If the returns are deliberate (and correct), then the preceding `WT_ERR_MSG` calls are wrong.

My suspicion is the code preceding the [general resource allocation, here](https://github.com/wiredtiger/wiredtiger/blob/develop/src/log/log.c#L2303-L2316) could be rewritten to not need a jump to the `err:` lable at all, but I didn't do that work. It should be possible, though, there's just a bit of trickery around the possible failure call to `__wt_fs_directory_list_free()`.

Reply:
I think this change is okay but also benign, happy to wait for @sueloverso to get back and share her opinion.